src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -388,19 +428,32 @@ public Response corsPreflightQuery(@HeaderParam("Access-Control-Request-Headers"
 	@GET
 	@Produces(MediaType.APPLICATION_JSON + "; charset=UTF-8")
 	@Path(QUERY_URL)
-	public Response query(@QueryParam("query") String json) throws Exception
+	public Response query(@Context HttpHeaders httpHeaders, @QueryParam("query") String json) throws Exception
 	{
-		return get(json);
+		return get(httpHeaders, json);
 	}
 
 	@POST
 	@Produces(MediaType.APPLICATION_JSON + "; charset=UTF-8")
 	@Path(QUERY_URL)
-	public Response get(String json) throws Exception
+	public Response get(@Context HttpHeaders httpHeaders, String json) throws Exception
 	{
 		checkNotNull(json);
 		logger.debug(json);
 
+		ScopeManager scopeManager = tracer.scopeManager();
```

Comment:
Is it possible to extract this piece of code into a separate method so we don't copy-paste it? It looks for me that it's a method that accepts `HttpHeaders` and `Tracer` objects, sets up Thread reporter and returns `Scope` object back. IntelliJ IDEA can help with refactoring (copy the code, then in main menu Refactor -> Extract -> Method)